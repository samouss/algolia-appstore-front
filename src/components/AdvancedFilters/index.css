.AdvancedFilters {
  display: flex;
  flex-direction: column;
  margin-top: 1vh;
}

.AdvancedFilters__Content {
  align-items: flex-start;
  margin-bottom: 1vh;
  display: none;

  @media (--tablet) {
    display: flex;
    flex-direction: row;
    margin-bottom: 0;
  }
}

.AdvancedFilters__Content--visible {
  display: flex;
  flex-direction: column;
}

.AdvancedFilters__Link {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 25px;
  color: #b9b9b9;
  font-size: 0.8rem;

  @media (--tablet) {
    display: none;
  }
}
