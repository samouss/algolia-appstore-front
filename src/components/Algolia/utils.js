import escape from 'lodash.escape';

export const highlightPreTag = '__algolia__highlight__pre__tag__';
export const highlightPostTag = '__algolia__highlight__post__tag__';

export const escapeAndReplaceTag = value => {
  return escape(value)
    .replace(highlightPreTag, '<em>')
    .replace(highlightPostTag, '</em>');
};
