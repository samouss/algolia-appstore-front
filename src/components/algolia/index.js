export { default as Provider } from './Provider';
export { default as withPaginateResults } from './withPaginateResults';
