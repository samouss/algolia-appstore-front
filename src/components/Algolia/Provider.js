import { Component, Children } from 'react';
import PropTypes from 'prop-types';
import algoliasearch from 'algoliasearch';
import algoliasearchHelper from 'algoliasearch-helper';
import { highlightPreTag, highlightPostTag } from './utils';

export const ContextTypes = {
  algoliaClient: PropTypes.object.isRequired,
  algoliaHelper: PropTypes.object.isRequired,
};

class Provider extends Component {
  constructor(props) {
    super(props);

    this.client = algoliasearch(props.appId, props.apiKey);
    this.helper = algoliasearchHelper(this.client, props.indexName, props.options);
  }

  getChildContext() {
    return {
      algoliaClient: this.client,
      algoliaHelper: this.helper,
    };
  }

  componentDidMount() {
    this.helper
      .setQueryParameter('highlightPreTag', highlightPreTag)
      .setQueryParameter('highlightPostTag', highlightPostTag)
      .search();
  }

  render() {
    return Children.only(this.props.children);
  }
}

Provider.propTypes = {
  appId: PropTypes.string.isRequired,
  apiKey: PropTypes.string.isRequired,
  indexName: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  options: PropTypes.object,
};

Provider.defaultProps = {
  options: {},
};

Provider.childContextTypes = ContextTypes;

export default Provider;
