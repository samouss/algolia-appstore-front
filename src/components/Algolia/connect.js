import React from 'react';
import PropTypes from 'prop-types';
import { getDisplayName } from 'core/utils';
import { ContextTypes } from './Provider';

export const ConnectPropTypes = {
  helper: PropTypes.object.isRequired,
};

export const connect = WrappedComponent => {
  const Connect = (props, context) => (
    <WrappedComponent
      {...props}
      helper={context.algoliaHelper}
    />
  );

  Connect.displayName = getDisplayName(
    WrappedComponent,
    'connect',
  );

  Connect.contextTypes = ContextTypes;

  return Connect;
};

export default connect;
