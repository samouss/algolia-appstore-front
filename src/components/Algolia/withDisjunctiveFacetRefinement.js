import React, { Component } from 'react';
import { getDisplayName } from 'core/utils';
import { ContextTypes } from './Provider';

const withDisjunctiveFacetRefinement = ({
  facet,
} = {}) => WrappedComponent => {
  class WithDisjunctiveFacetRefinement extends Component {

    constructor(props) {
      super(props);

      this.onChange = this.onChange.bind(this);
    }

    onChange(facetValue, isRefined) {
      const { algoliaHelper } = this.context;

      if (isRefined) {
        return algoliaHelper
          .removeDisjunctiveFacetRefinement(facet)
          .search();
      }

      algoliaHelper
        .removeDisjunctiveFacetRefinement(facet)
        .addDisjunctiveFacetRefinement(facet, facetValue)
        .search();
    }

    render() {
      return (
        <WrappedComponent
          {...this.props}
          onChange={this.onChange}
        />
      );
    }

  }

  WithDisjunctiveFacetRefinement.displayName = getDisplayName(
    WrappedComponent,
    'withDisjunctiveFacetRefinement',
  );

  WithDisjunctiveFacetRefinement.contextTypes = ContextTypes;

  return WithDisjunctiveFacetRefinement;
};

export default withDisjunctiveFacetRefinement;
