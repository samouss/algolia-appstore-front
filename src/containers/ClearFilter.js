import flowRight from 'lodash.flowright';
import { withRefinements, withClearRefinements } from 'components/Algolia';
import ClearList from 'components/ClearList';

export default flowRight(
  withRefinements,
  withClearRefinements,
)(ClearList);
