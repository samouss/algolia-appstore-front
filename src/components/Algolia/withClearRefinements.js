import flowRight from 'lodash.flowright';
import React, { Component } from 'react';
import { getDisplayName } from 'core/utils';
import connect, { ConnectPropTypes } from './connect';

export const withClearRefinements = WrappedComponent => {
  class WithClearRefinements extends Component {

    constructor(props) {
      super(props);

      this.onClear = this.onClear.bind(this);
    }

    onClear() {
      const { helper } = this.props;

      helper
        .clearRefinements()
        .search();
    }

    render() {
      const { helper, ...props } = this.props;

      return (
        <WrappedComponent
          {...props}
          onClear={this.onClear}
        />
      );
    }

  }

  WithClearRefinements.displayName = getDisplayName(
    WrappedComponent,
    'withClearRefinements',
  );

  WithClearRefinements.propTypes = ConnectPropTypes;

  return WithClearRefinements;
};

export default flowRight(
  connect,
  withClearRefinements,
);
