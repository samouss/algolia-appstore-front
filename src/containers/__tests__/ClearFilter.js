import React from 'react';
import { shallow } from 'enzyme';
import { withRefinements, withClearRefinements } from 'components/Algolia';
import ClearFilter from '../ClearFilter';

jest.mock('components/Algolia', () => ({
  withRefinements: jest.fn(Component => props => (
    <Component
      {...props}
      refinements={[]}
    />
  )),
  withClearRefinements: jest.fn(Component => props => (
    <Component
      {...props}
      onClear={() => {}}
    />
  )),
}));

describe('<ClearFilter />', () => {
  it('expect to render', () => {
    const component = shallow(<ClearFilter />).shallow();

    expect(component).toMatchSnapshot();
  });

  describe('withRefinements', () => {
    it('expect to be called', () => {
      expect(withRefinements).toHaveBeenCalled();
    });
  });

  describe('withClearRefinements', () => {
    it('expect to be called', () => {
      expect(withClearRefinements).toHaveBeenCalled();
    });
  });
});
