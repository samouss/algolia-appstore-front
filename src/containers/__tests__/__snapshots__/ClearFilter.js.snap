// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClearFilter /> expect to render 1`] = `
<ClearList
  onClear={[Function]}
  refinements={Array []}
/>
`;
